File: 800/C++/p13.cpp
#include<bits/stdc++.h>
using namespace std;

void solve() {
    int n;
    cin>>n;
    vector<int>arr(n);
    for(int i = 0; i < n; i++)  cin>>arr[i];

    int maxi = *max_element(arr.begin(), arr.end());

    vector<int>first, second;

    for(int i = 0; i < n; i++) {
        if(arr[i] == maxi)  second.push_back(arr[i]);
        else    first.push_back(arr[i]);
    }

    // if all elements are equal to maxi
    if(first.size() == 0)   cout<<-1<<endl;
    else {
        cout<<first.size()<<" "<<second.size()<<endl;
        for(auto it:first)  cout<<it<<" ";
        cout<<endl;
        for(auto it:second) cout<<it<<" ";
        cout<<endl;
    }
}

int main() {
    int t;
    cin>>t;
    while(t--)  solve();
    return 0;
}

File: 800/Java/P13.java
import java.util.*;

public class P13 {
    static void solve(Scanner scanner) {
        int n = scanner.nextInt();
        long []arr = new long[n];
        for(int i = 0; i < n; i++)  arr[i] = scanner.nextLong();

        long maxi = Arrays.stream(arr).max().getAsLong();

        ArrayList<Long>first = new ArrayList<>();
        ArrayList<Long>second = new ArrayList<>();

        for(int i = 0; i < n; i++) {
            if(arr[i] == maxi)  second.add(arr[i]);
            else    first.add(arr[i]);
        }

        if(first.size() == 0)   System.out.println(-1);
        else {
            System.out.print(first.size() + " " + second.size() + "\n");
            for(int i = 0; i < first.size(); i++)   System.out.print(first.get(i) + " ");
            System.out.println();
            for(int i = 0; i < second.size(); i++)  System.out.print(second.get(i) + " ");
            System.out.println();
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt();
        while(t-- > 0)  solve(scanner);
        scanner.close();
    }
}